[SpeedDial] unresolved dependencies in TypeScript when following docs

<!--- Provide a general summary of the issue in the Title above -->

<!--
    Thank you very much for contributing to Material-UI by creating an issue! ❤️
    To avoid duplicate issues we ask you to check off the following list.
-->

<!-- Checked checkbox should look like this: [x] -->
- [x] This is a v1.x issue (v0.x is no longer maintained).
- [x] I have searched the [issues](https://github.com/mui-org/material-ui/issues) of this repository and believe that this is not a duplicate.

## Expected Behavior
<!---
    If you're describing a bug, tell us what should happen.
    If you're suggesting a change/improvement, tell us how it should work.
-->
A working speed dial according to the docs when using TypeScript.

## Current Behavior
<!---
    If describing a bug, tell us what happens instead of the expected behavior.
    If suggesting a change/improvement, explain the difference from current behavior.
-->
Currently, I get the following error whenever I try to use the speed dial without having `@material-ui/icons` installed:
`Module not found: Can't resolve '@material-ui/icons/Add' in '../node_modules/@material-ui/lab/SpeedDialIcon'`

The documentation does not specify that this is required, and it shouldn't be a requirement in my opinion, as this enforces users to use this dependency instead of using something like font awesome.

## Steps to Reproduce (for bugs)
<!---
    Provide a link to a live example (you can use codesandbox.io) and an unambiguous set of steps to reproduce this bug.
    Include code to reproduce, if relevant (which it most likely is).

    This codesandbox.io template _may_ be a good starting point:
    https://codesandbox.io/s/github/mui-org/material-ui/tree/master/examples/create-react-app


    If YOU DO NOT take time to provide a codesandbox.io reproduction, should the COMMUNITY take time to help you?

-->

1. create a react app with TypeScript (`npx create-react-app app --scripts-version=react-scripts-ts`)
2. add `@material-ui/core` and `@material-ui/lab` as dependencies
3. try to use the `SpeedDialIcon` component

Codesandbox example: https://codesandbox.io/s/n36wk5xzkp
This does not give the exact same error, but it definitely points towards the same issue.

## Context
<!---
    How has this issue affected you? What are you trying to accomplish?
    Providing context helps us come up with a solution that is most useful in the real world.
-->
Wanting to use the speed dial icons without having `material-ui/icons` as a necessary dependency, as we use generic components for pre-bootstrapped functionality across our application.

## Your Environment
<!--- Include as many relevant details about the environment with which you experienced the bug. -->

| Tech         | Version |
|--------------|---------|
| @material-ui/core  | v1.3.1  |
| @material-ui/lab | v1.0.0-alpha.5 |
| React        | v16.4.1  |
| TypeScript | v2.9.2 |

--- a/packages/material-ui-lab/package.json
+++ b/packages/material-ui-lab/package.json
@@ -33,7 +33,6 @@
     "release": "yarn build && npm publish build"
   },
   "peerDependencies": {
-    "@material-ui/icons": "^1.0.0",
     "@material-ui/core": "^1.0.0",
     "react": "^16.3.0",
     "react-dom": "^16.3.0"

--- a/packages/material-ui-lab/src/SpeedDialIcon/SpeedDialIcon.js
+++ b/packages/material-ui-lab/src/SpeedDialIcon/SpeedDialIcon.js
@@ -2,7 +2,7 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import { withStyles } from '@material-ui/core/styles';
-import AddIcon from '@material-ui/icons/Add';
+import AddIcon from '../internal/svg-icons/Add';
 
 const styles = theme => ({
   root: {

--- a/packages/material-ui-lab/src/internal/svg-icons/.eslintrc.js
+++ b/packages/material-ui-lab/src/internal/svg-icons/.eslintrc.js
@@ -1,0 +1,6 @@
+module.exports = {
+  rules: {
+    'import/no-mutable-exports': 'off',
+    'max-len': 'off',
+  },
+};

--- a/packages/material-ui-lab/src/internal/svg-icons/Add.js
+++ b/packages/material-ui-lab/src/internal/svg-icons/Add.js
@@ -1,0 +1,17 @@
+import React from 'react';
+import pure from 'recompose/pure';
+import SvgIcon from '@material-ui/core/SvgIcon';
+
+/**
+ * @ignore - internal component.
+ */
+let Add = props => (
+  <SvgIcon {...props}>
+    <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z" />
+  </SvgIcon>
+);
+
+Add = pure(Add);
+Add.muiName = 'SvgIcon';
+
+export default Add;

